core/src/main/resources/xtdb/sql/parser/sql.ebnf

```diff
@@ -495,6 +501,10 @@ interval_type
     : 'INTERVAL' interval_qualifier
     ;
 
+duration_type 
+    : 'DURATION'
```

Comment:
Not sure if we really expose `DURATION` much to SQL - if we want to cast to/from duration we will need some concept of a duration type, so added one here as a possible cast target (will need similar for #3196)